app/src/main/java/org/simple/clinic/facility/change/FacilityListItem.kt

```diff
@@ -27,6 +61,42 @@ sealed class FacilityListItem {
       data class WithStreet(val street: String, val district: String, val state: String) : Address()
       data class WithoutStreet(val district: String, val state: String) : Address()
     }
+
+    override fun layoutResId(): Int {
+      return R.layout.list_facility_selection_option
+    }
+
+    override fun render(holder: ViewHolderX, subject: Subject<FacilityItemClicked>) {
+      when (name) {
+        is Highlighted -> {
+          val highlightedName = SpannableStringBuilder(name.text)
+          val highlightColor = ContextCompat.getColor(holder.itemView.context, R.color.facility_search_query_highlight)
+          highlightedName.setSpan(BackgroundColorSpan(highlightColor), name.highlightStart, name.highlightEnd, Spannable.SPAN_EXCLUSIVE_EXCLUSIVE)
+          holder.facilityNameTextView.text = highlightedName
+        }
+        is Plain -> {
+          holder.facilityNameTextView.text = name.text
+        }
+      }.exhaustive()
+
+      holder.facilityAddressTextView.text = when (address) {
+        is Address.WithStreet -> {
+          holder.itemView.resources.getString(
+              R.string.registrationfacilities_facility_address_with_street,
+              address.street,
+              address.district,
+              address.state)
+        }
+        is Address.WithoutStreet -> {
+          holder.itemView.resources.getString(
+              R.string.registrationfacilities_facility_address_without_street,
+              address.district,
+              address.state)
+        }
+      }
+
+      holder.itemView.setOnClickListener { subject.onNext(FacilityItemClicked(facility)) }
```

Comment:
It's not related to this PR, but one thing I have been thinking about is. Isn't it not recommended to have click listeners when we are binding the view to recycler view since it's called for every item? 🤔

Reply:
The alternative is to track the current item in a mutable variable and update it when binding, which adds additional complexity. We made this call to make the views as stateless as possible. Eventually, when Jetpack Compose becomes production ready, this will make it easier for us to migrate if we would like to.

Reply:
or can't we get the views in `onCreateView`? and get the item for the position when the item is clicked. WDYT?

Reply:
I didn't understand this. `onCreateView` is used where?

Reply:
Sorry, `onCreateViewHolder`.  Since that is only called when we are creating the `ViewHolder`, that seems like a good option.

Reply:
Ah. But that would mean then that the item view needs to maintain a reference to the entire list in some way so it can look up the item when a click happens.

Reply:
When using `ListAdapter` we can just call `getItem()`, and `ViewHolder` should have the adapter position available in click listeners.

Reply:
That's not possible here since we have a generic view holder that wraps an `Item` generic type. These types do not have access to the parent adapter so they cannot invoke that method.